pandas/core/indexes/datetimelike.py

```diff
@@ -595,7 +596,12 @@ def _maybe_cast_listlike_indexer(self, keyarr):
         try:
             res = self._data._validate_listlike(keyarr, allow_object=True)
         except (ValueError, TypeError):
-            res = com.asarray_tuplesafe(keyarr)
+            if not isinstance(keyarr, ExtensionArray):
+                # e.g. we don't want to cast DTA to ndarray[object]
+                res = com.asarray_tuplesafe(keyarr)
```

Comment:
should we just have ExtensionArray pass thru asarray_tuplesafe (would need to change the return signature), not sure if this is used enough to warrant but a thought.